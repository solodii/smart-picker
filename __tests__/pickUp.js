var createPicker = require('../src/index').createPicker;
var mocks = require('../mocks');

describe('pickUp', function() {
  var picker = createPicker(mocks.schema);

  it('should return an object if target is not an object', function() {
    expect(picker.pickUp()).toEqual({});
    expect(picker.pickUp(null)).toEqual({});
    expect(picker.pickUp(0)).toEqual({});
    expect(picker.pickUp('')).toEqual({});
  });

  it('should pickUp queried object properties', function() {
    expect(
      picker.pickUp(mocks.target)
    ).toMatchSnapshot();
  });
});
