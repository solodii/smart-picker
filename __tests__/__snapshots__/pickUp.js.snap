// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pickUp should pickUp queried object properties 1`] = `
Object {
  "a_alias": "a_prop",
  "d_alias": "d_prop",
  "f": Object {
    "g": "g_prop",
    "h": "h_prop",
  },
  "length": 3,
}
`;
